src/agent/onefuzz-supervisor/src/coordinator.rs

```diff
@@ -95,11 +95,18 @@ pub enum TaskState {
     WaitJob,
 }
 
-#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
-pub struct TaskSearch {
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CanScheduleRequest {
+    version: String,
     task_id: Uuid,
 }
 
+#[derive(Clone, Copy, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CanSchedule {
+    pub allowed: bool,
+    pub work_stopped: bool,
+}
```

Comment:
can we add an explicit field for the version check

Reply:
Not doing this, per offline discussion, since we'll have the `can_schedule()` check stay ignorant of versions / if it should stop.